docs/freqtrade-101.md

```diff
@@ -96,11 +103,60 @@ class MyStrategy(IStrategy):
+
+However, there can be a range of scenarios where generating a signal does not always create a trade order. These include:
+
+- not enough remaining stake to buy an asset, or funds in your wallet to sell an asset (including any fees)
+- not enough open slots for a new trade to be opened
+- there is already an open trade for a pair (Freqtrade cannot stack positions - however it can [adjust existing positions](strategy-callbacks.md#adjust-trade-position))
+- if an entry and exit signal is present on the same candle, they cancel each other out and no order will be raised
+- the strategy actively rejects the trade order due to logic you specify by using one of the relevant [entry](strategy-callbacks.md#trade-entry-buy-order-confirmation) or [exit](strategy-callbacks.md#trade-exit-sell-order-confirmation) callbacks
+
+Read through the [strategy customization](strategy-customization.md) documentation for more details.
+
+## Backtesting and forward testing
+
+Strategy development can be a long and frustrating process, as turning our human "gut instincts" into a working computer-controlled
+("algo") strategy is not always straightforward.
+
+Therefore a strategy should be tested to verify that it is going to work as intended.
 
+Freqtrade has two testing modes:
+
+- **backtesting**: using historical data that you [download from an exchange](data-download.md), backtesting is a quick way to assess performance of a strategy. However, it can be very easy to distort results so a strategy will look a lot more profitable than it really is. Check the [backtesting documentation](backtesting.md) for more information. 
+- **dry run**: often referred to as `forward testing`, dry runs use real time data from the exchange. However, any signals that would result in trades are tracked as normal by Freqtrade, but do not have any trades opened on the exchange itself. Forward testing runs in real time, so whilst it takes longer to get results it is a much more reliable indicator of **potential** performance than backtesting. 
+
+Dry runs are enabled by setting `dry_run` to true in your [configuration](configuration.md#using-dry-run-mode).
+
+!!! Warning "Backtests can be very inaccurate"
+    There are many reasons why backtest results will not match reality. Please check the [backtesting assumptions](backtesting.md#assumptions-made-by-backtesting) and [common strategy mistakes](strategy-customization.md#common-mistakes-when-developing-strategies) documentation.
+
+!!! Warning "Public strategies can have significant issues"
+    There are many websites listing strategies with impressive backtest results. Do not assume these results are achieveable or realistic.
+
+??? Hint "Useful commands"
+    Freqtrade includes two useful commands to check for basic flaws in strategies: [lookahead-analysis](lookahead-analysis.md) and [recursive-analysis](recursive-analysis.md).
+
+!!! Note "Always dry run first!"
```

Comment:
I found that putting several boxes in sequence reduces the likelyhood of readers actually reading the content with every additional box.

maybe we can combine the 2 warnings into one bigger box?

Reply:
Done